### PR TITLE
docs: Initial Crate level RustDoc for oreboot_nezha_bt0

### PR DESCRIPTION
**Pull request by bwerthmann**
Initially, include path was `../../README.md`. However, in this configuration Cargo's change detection failed to invalidate artifact cache when the included document changes.

```rust
#![doc = include_str!("../../README.md")]
```

The fix is to:

1. make a symbolic link from `oreboot_nezha_bt0` src dir to document

    README.md -> ../../README.md

2. Set Crate level doc include path to symbolic link `README.md`

```rust
#![doc = include_str!("README.md")]
```

3. Build and Open RustDoc for `oreboot_nezha_bt0`

```bash
cd oreboot
cd src/mainboard/sunxi/nezha/bt0
cargo doc --document-private-items --open
```

```console
   Compiling oreboot-nezha-bt0 v0.1.0 (oreboot/src/mainboard/sunxi/nezha/bt0)
warning: the feature `int_abs_diff` has been stable since 1.60.0 and no longer requires an attribute to enable
 --> src/soc/src/lib.rs:2:12
  |
2 | #![feature(int_abs_diff)]
  |            ^^^^^^^^^^^^
  |
  = note: `#[warn(stable_features)]` on by default

warning: type alias `PGUART` is never used
  --> src/soc/src/sunxi/d1/uart.rs:19:6
   |
19 | type PGUART = (PG17<Function<7>>, PG18<Function<7>>);
   |      ^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: `oreboot-soc` (lib) generated 2 warnings
 Documenting oreboot-nezha-bt0 v0.1.0 (/home/ben/ofw/oreboot/src/mainboard/sunxi/nezha/bt0)
warning: this URL is not a hyperlink
 --> src/mainboard/sunxi/nezha/bt0/src/main.rs:1:1
  |
1 | #![doc = include_str!("README.md")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use an automatic link instead: `<https://github.com/orangecms/linux/tree/5.19-smaeul-plus-dts>`
  |
  = note: bare URLs are not automatically turned into clickable links
  = note: `#[warn(rustdoc::bare_urls)]` on by default

```

4. Make a change to README.md
5. Build docs again and note that Cargo invalidates caches, and builds new Documentation artifacts

Signed-off-by: Ben Werthmann <ben.werthmann@gmail.com>

### CODE REVIEW

**Comment by bwerthmann** [comment]
@orangecms PTAL.

File: src/mainboard/sunxi/nezha/README.md [hunk 1]
```diff
@@ -37,8 +37,8 @@ and to be found within Linux in the directory `arch/riscv/boot/dts/allwinner/`.
 We currently do not patch it at runtime, so you need to hardcode your board's
 memory size into it.
 
-For a work-in-progress Linux 5.19 (not all patches are upstream yet), see:
-https://github.com/orangecms/linux/tree/5.19-smaeul-plus-dts
+Not all patches are upstream yet. [work-in-progress Linux 5.19](
+https://github.com/orangecms/linux/tree/5.19-smaeul-plus-dts/arch/riscv/boot/dts/allwinner)
```
Comment (orangecms) [inline src/mainboard/sunxi/nezha/README.md:41]: More than just the DTS is needed, so the reference was to the branch.

How do you feel about keeping the phrasing as is, adding the Markdown link formatting, and giving the link a title that is the URL again, so it'll be renderd in Rust docs just like on GitHub?
We could improve the wording thereafter.

**Review by orangecms — CHANGES_REQUESTED** [review]

File: src/mainboard/sunxi/nezha/README.md [hunk 2]
```diff
@@ -30,15 +30,19 @@ make flashwithpayload \
   DTB="$_LINUX_BOOT/dts/allwinner/sun20i-d1-$_BOARD.dtb"
 ```
 
-**Note**
+#### Notes on Device Tree Blob (DTB)
```
Comment (bwerthmann) [inline src/mainboard/sunxi/nezha/README.md:33]: @orangecms WDYT?

**Comment by bwerthmann** [comment]
Changes to doc content moved to #616.

File: src/mainboard/sunxi/nezha/README.md (hunk 2 above)
Comment (bwerthmann) [reply src/mainboard/sunxi/nezha/README.md]: #616

**Comment by bwerthmann** [comment]
@orangecms should be :green_circle: now.

**Review by orangecms — APPROVED** [review]

**Pull request merged**